rust/noosphere/src/sphere/builder/recover.rs

```diff
@@ -0,0 +1,144 @@
+use std::sync::Arc;
+
+use anyhow::{anyhow, Result};
+use cid::Cid;
+use noosphere_core::{
+    authority::{generate_capability, generate_ed25519_key, Author, Authorization, SphereAbility},
+    context::{
+        SphereAuthorityRead, SphereContext, SphereSync, SyncExtent, SyncRecovery, AUTHORIZATION,
+        GATEWAY_URL, IDENTITY, USER_KEY_NAME,
+    },
+    data::Did,
+};
+use noosphere_storage::KeyValueStore;
+use tokio::sync::Mutex;
+use ucan::{builder::UcanBuilder, crypto::KeyMaterial};
+
+use crate::{
+    key::KeyStorage,
+    sphere::{generate_db, SphereContextBuilder, SphereContextBuilderArtifacts},
+};
+
+pub async fn recover_a_sphere(
+    builder: SphereContextBuilder,
+    sphere_identity: Did,
+) -> Result<SphereContextBuilderArtifacts> {
+    // 1. Check to make sure the local key is available
+
+    let key_storage = builder.require_key_storage()?;
+    let local_key_name = builder.require_key_name()?.to_owned();
+    let local_key_did = Did(key_storage
+        .require_key(&local_key_name)
+        .await?
+        .get_did()
+        .await?);
+
+    // 2. Backup existing database, moving it to a new location
+    // NOTE: Database backups not available on wasm targets
+
+    let storage_path = builder.require_storage_path()?.to_owned();
+
+    #[cfg(not(target_arch = "wasm32"))]
```

Comment:
This could be useful as well in #633 e.g. as a `Movable` or `Backupable` storage trait. Valuable to handle this in only one spot, with getting it wrong having high costs. 

We don't yet have a concrete backup strategy in the db-recovery sense, but storing these somewhere is the first step, also to be used during db migrations. I can imagine something (for power users in an exceptional state) like  `orb sphere backups revert 123456`.

Nothing actionable here, but lets consolidate on one way of handling backups (will synchronize #633), and follow up on any strategies for eviction or recovery (not in the sense of this PR, but in the "i tried migrating to an experimental DB and want to roll back" sense).

Reply:
Yah, if you want to take a stab at factoring this out, that all makes sense to me.